### PR TITLE
[new-feature](complex-type) support read nested parquet and orc file with complex type

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/exec/format/parquet/vparquet_column_reader.cpp [hunk 1]
```diff
@@ -150,31 +138,23 @@ Status ParquetColumnReader::create(io::FileReaderSPtr file, FieldSchema* field,
...
-            auto child_type = field->children[i].type.type;
-            if (child_type == TYPE_ARRAY || child_type == TYPE_MAP || child_type == TYPE_STRUCT) {
-                return Status::InternalError(
-                        "Struct does not support nested complex type in column {}", field->name);
-            }
             std::unique_ptr<ParquetColumnReader> child_reader;
             RETURN_IF_ERROR(create(file, &field->children[i], row_group, row_ranges, ctz, io_ctx,
                                    child_reader, max_buf_size));
             child_reader->set_nested_column();
             child_readers.emplace_back(std::move(child_reader));
```
Comment (airborne12) [inline be/src/vec/exec/format/parquet/vparquet_column_reader.cpp:150]: for efficient memory allocate, may be the following is better
```
child_readers.reserve(field->children.size());
for (auto& child : field->children) {
  auto child_reader = std::make_unique<ParquetColumnReader>();
  RETURN_IF_ERROR(create(file, &field->children[i], row_group, row_ranges, ctz, io_ctx,
                                    child_reader, max_buf_size));  child_reader->set_nested_column();
  child_readers.emplace_back(std::move(child_reader));
}
```

**Comment by AshinGau** [comment]
LGTM

**Review by AshinGau — DISMISSED** [review]

File: be/src/vec/exec/format/parquet/vparquet_column_reader.cpp [hunk 2]
```diff
@@ -123,25 +123,13 @@ Status ParquetColumnReader::create(io::FileReaderSPtr file, FieldSchema* field,
...
-                    "Array does not support nested map/struct type in column {}", field->name);
-        }
         RETURN_IF_ERROR(create(file, &field->children[0], row_group, row_ranges, ctz, io_ctx,
                                element_reader, max_buf_size));
         element_reader->set_nested_column();
-        ArrayColumnReader* array_reader = new ArrayColumnReader(row_ranges, ctz, io_ctx);
+        auto array_reader = ArrayColumnReader::create_unique(row_ranges, ctz, io_ctx);
         RETURN_IF_ERROR(array_reader->init(std::move(element_reader), field));
-        reader.reset(array_reader);
+        reader.reset();
```
Comment (yiguolei) [inline be/src/vec/exec/format/parquet/vparquet_column_reader.cpp:131]: should reset to array_reader.release()

Comment (amorynan) [reply be/src/vec/exec/format/parquet/vparquet_column_reader.cpp]: done

**Comment by amorynan** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
run clickbench

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.6 seconds
 stream load tsv:          514 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162319578 Bytes

**Pull request merged**